examples/chip-tool/commands/clusters/WriteAttributeCommand.h

```diff
@@ -105,6 +110,38 @@ class WriteAttribute : public ModelCommand, public chip::app::WriteClient::Callb
         return mWriteClient->SendWriteRequest(device->GetSecureSession().Value());
     }
 
+    template <class T>
+    CHIP_ERROR SendGroupCommand(chip::GroupId groupId, chip::FabricIndex fabricIndex, chip::NodeId senderNodeId,
+                                chip::ClusterId clusterId, chip::AttributeId attributeId, const T & value)
+    {
+
+        chip::app::AttributePathParams attributePathParams;
+        attributePathParams.mClusterId   = clusterId;
+        attributePathParams.mAttributeId = attributeId;
+
+        chip::Messaging::ExchangeManager * exchangeManager = chip::app::InteractionModelEngine::GetInstance()->GetExchangeManager();
+
+        ChipLogDetail(chipTool, "Sending Write Attribute to Group %u, on Fabric %x, for cluster %u with attributeId %u", groupId,
+                      fabricIndex, clusterId, attributeId);
+
+        auto writeClient = chip::Platform::MakeUnique<chip::app::WriteClient>(exchangeManager, this, mTimedInteractionTimeoutMs);
+        VerifyOrReturnError(writeClient != nullptr, CHIP_ERROR_NO_MEMORY);
+        ReturnErrorOnFailure(writeClient->EncodeAttribute(attributePathParams, value, mDataVersion));
```

Comment:
All of this logic is already present in `chip::Controller::WriteAttribute` - why are we duplicating that here?

@vivien-apple why isn't this file just using those functions, instead of duplicating all of those bits here in this file?

Reply:
Legacy stuff most probably. The goal was to add the feature to chip-tool so that interested party could start sending group commands/attribute write with it asap not to revisit how chip-tool works...